sovrin_client/agent/agent.py

```diff
@@ -190,9 +196,18 @@ def __init__(self,
                  attrRepo=None):
 
         Agent.__init__(self, name, basedirpath, client, port, loop=loop)
-        self._wallet = wallet or Wallet(name)
+
+        self.config = getConfig(basedirpath)
+
+        self._restoreLastActiveWallet()
+
+        if wallet:
+            self._wallet = wallet
```

Comment:
I think we can remove `self._wallet` altogether. We can just use `self.wallet`. Since we are always relying on one wallet, either by restoring from file or if wallet is passed. `self._wallet` seems unnecessary.